src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -115,11 +115,12 @@ def _nailgunned_stdio(self, sock):
            (ChunkType.STDOUT, ChunkType.STDERR),
            None,
            (stdout_isatty, stderr_isatty)
-         ) as ((stdout, stderr), writer),\
-         NailgunStreamStdinReader.open(sock, stdin_isatty) as stdin,\
-         stdio_as(stdout=stdout, stderr=stderr, stdin=stdin):
+         ) as ((stdout_fd, stderr_fd), writer),\
+         NailgunStreamStdinReader.open(sock, stdin_isatty) as stdin_fd,\
+         stdio_as(stdout_fd=stdout_fd, stderr_fd=stderr_fd, stdin_fd=stdin_fd):
       # N.B. This will be passed to and called by the `DaemonExiter` prior to sending an
       # exit chunk, to avoid any socket shutdown vs write races.
+      stdout, stderr = sys.stdout, sys.stderr
```

Comment:
this should probably move inside the `def finalizer` block?

Reply:
It's intentionally outside, in order to capture (alias) `sys.std*` as it is inside the block. I'm not 100% certain where the finalizer will be called?